### PR TITLE
Ts protein summary

### PR DESCRIPTION
**Pull request by gustavo-salazar**
TypeScript and visual framework migration for the Protein summary page. i.e. the landing page when choosing a protein.

Files migrated:
* `src/components/Protein/Summary`
* `src/components/File`
* `src/components/Matches/FileExporter`
* `src/components/Protein/Isoforms/Selector`
* `src/components/Protein/Isoforms/Viewer`
* `src/components/Protein/Length`
* `src/components/Protein/Sequence/DownloadButton`
* `src/components/Protein/Sequence/HmmerButton`
* `src/components/Protein/Sequence/InterProScanButton`
* `src/components/Protein/Species`
* `src/utils/sequence`

New files created to organise the code:
* `src/components/File/FIleButton/`
* `src/components/File/TooltipContent/`

#### Other Changes
Hiding the HmmerButton until Hmmer is more stable

#### Dependencies updates
The update of `postcss` deprecates the use of `@nest` so it gets removed from our code base.

---

**Note:** Given the dependencies update, you will need to run `npm install` to test this.

### CODE REVIEW

**Comment by gustavo-salazar** [comment]
@typhainepl the last commit solves the dashed line in the left menu, adds a subtitle to the protein viewer with the canonical isoform, and moves the entry herarchy to the table on top.

Unfortunately, I haven't been able to solve the issue with the printing, this is actually a bug in the live website, and is not related to this particular PR. Please open an issue for it, I'll see what I can do. FYI, the functionality is working on firefox 
🤷🏽

**Comment by typhainepl** [comment]
Looking better :)

Is it on purpose that:
1. the isoform doesn't have the labels on the domains?
2. the isoform/canonical names are with a blue background?

**Comment by gustavo-salazar** [comment]
> the isoform doesn't have the labels on the domains?

Isoforms data doesn't have short names, which is what we show in there. I can put the accession or the long name instead.

> the isoform/canonical names are with a blue background?

Yeah, "on purpose", I'll change it to something nicer

**Comment by typhainepl** [comment]
Looking better :)

File: src/components/Protein/Summary/index.tsx [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+                          'vf-button--secondary',
+                          'vf-button--sm',
+                          'comparison-button',
+                          'icon',
+                          'icon-common',
+                          { disabled: !isoform }
+                        )}
+                        disabled={!isoform}
+                        dataIcon={'\uF0DB'}
+                        tooltip="View in comparison mode"
```
Comment (matthiasblum) [inline src/components/Protein/Summary/index.tsx:236]:
```suggestion
                        tooltip="Compare side-by-side with canonical sequence"
```

File: src/components/Protein/Summary/index.tsx [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+          </table>
+        </div>
+        <div className={css('vf-stack')}>
+          <section>
+            <section>
+              <h5>External Links</h5>
+              <ul className={css('no-bullet')}>
+                <li>
+                  <UniProtLink id={metadata.accession} className={css('ext')}>
+                    View {metadata.accession} in UniProtKB
```
Comment (matthiasblum) [inline src/components/Protein/Summary/index.tsx:254]:
```suggestion
                    UniProt
```

(Technically, the service is UniProt. UniprotKB is the database of proteins.)

File: src/components/Matches/FileExporter/index.tsx [hunk 3]
```diff
@@ -1,52 +1,54 @@
...
+};
+const FileExporter = ({
+  description,
+  search = {},
+  count,
+  fileType,
+  primary,
+  secondary,
+  className,
+  focused = null,
```
Comment (matthiasblum) [inline src/components/Matches/FileExporter/index.tsx:49]: Exposing `label` (defined in props):

```suggestion
  label,
  focused = null,
```

File: src/components/Matches/FileExporter/index.tsx [hunk 4]
```diff
@@ -74,21 +76,10 @@ const FileExporter = (
           search.extra_fields ? `,${search.extra_fields}` : ''
         }`,
       }}
-      endpoint={(endpoint[primary] && endpoint[primary][secondary]) || primary}
+      endpoint={(endpoint?.[primary]?.[secondary] as string) || primary}
       minWidth={minWidth}
```
Comment (matthiasblum) [inline src/components/Matches/FileExporter/index.tsx:80]: Passing `label` to `File`:

```suggestion
      minWidth={minWidth}
      label={label}
```

File: src/components/File/index.tsx [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+  customLocationDescription?: Object;
+  downloadURL: typeof downloadURL;
+  fileType: SupportedExtensions;
+  count: number;
+  subset?: boolean;
+  name: string;
+  search?: Record<string, string>;
+  endpoint?: string;
+  className?: string;
+  minWidth?: number | string;
```
Comment (matthiasblum) [inline src/components/File/index.tsx:36]: Allowing custom label:

```suggestion
  minWidth?: number | string;
  label?: string;
```

File: src/components/Protein/Summary/index.tsx [hunk 6]
```diff
@@ -0,0 +1,336 @@
...
+                metadata.sequence,
+                metadata.id || ''
+              )}
+              title="Search protein with InterProScan"
+              minWidth={minWidth}
+            />
+            <DownloadButton
+              sequence={metadata.sequence}
+              accession={metadata.accession}
+            />
```
Comment (matthiasblum) [inline src/components/Protein/Summary/index.tsx:298]: Suggestions:

- remove the `<h5>` element above the TSV export button, and use a custom label for the button's title (requires changes I suggested to the `FileExporter` and `File` components.
- reorganize the buttons (makes more sense to have to two download buttons together) 
- change the title of the `IPSCanButton` from "Search protein with InterProScan" to "Search sequence with InterProScan"

```suggestion
            {/* <HmmerButton
              sequence={metadata.sequence}
              accession={metadata.accession}
              title="Search protein with HMMER"
              minWidth={minWidth}
            />
            <br /> */}
            <IPScanButton
              sequence={splitSequenceByChunks(
                metadata.sequence,
                metadata.id || ''
              )}
              title="Search sequence with InterProScan"
              minWidth={minWidth}
            />
            {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
            <label>
              <FileExporter
                description={{
                  main: { key: 'protein' },
                  protein: {
                    db: metadata.source_database,
                    accession: metadata.accession,
                  },
                  entry: { integration: 'all' },
                }}
                count={metadata.counters.entries as number}
                fileType="tsv"
                primary="entry"
                secondary="protein"
                minWidth={minWidth}
                label="Download matches (TSV)"
              />
            </label>
            <DownloadButton
              sequence={metadata.sequence}
              accession={metadata.accession}
            />
```

**Review by matthiasblum — APPROVED** [review]
Looks good to me.

I've made a few suggestions, but these are based personal preferences. Feel free to ignore them if you believe they are out-of-scope of this PR.

Another point: shouldn't we hide the main protein viewer when a user display an isoform matches?

File: src/components/File/index.tsx [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+};
+
+type State = {
+  url: string | null;
+  fileType: string | null;
+  subset: boolean | null;
+  ConnectedButton: typeof FileButton | null;
+  subpath: string | null;
```
[the remark this answers is not in the document]
Comment (gustavo-salazar) [reply src/components/File/index.tsx:44]: Because I'm initiating the state of those as `null` and TS will complain as `null!==undefined`

**Comment by gustavo-salazar** [comment]
> Another point: shouldn't we hide the main protein viewer when a user display an isoform matches?

Mhh, I'm unsure about this one. Yes it might be tidier that way , but there are tracks that are only available in the canonical (e.g. alphafold and residues) which makes me think keeping the canonical on the page can still be useful

**Pull request merged**